File: backend/src/validator/meet.py
from marshmallow import fields, Schema
from marshmallow.validate import Length

class MeetSchema(Schema):
    role = fields.Str(required=True, validate=Length(equal=24))
    date = fields.Date(required=True)
    dateFormat = fields.Str(required=True)
    hour = fields.Str(required=False)
    ubication = fields.Str(required=True)
    student = fields.Dict(required=True)
    attendance = fields.Bool(required=True)
    state = fields.Str(required=True)
    postulation = fields.Str(required=True, allow_none=True)

File: frontend/src/app/model/meet.ts
export interface Meet {
  _id: any;
  student: {
    nombre: String;
    correo: String;
    codigo: String;
  };
  postulation: String;
  state: String;
  ubication: String;
  date: Date;
  reason?: String;
  hour: String;
  role: String;
  dateFormat: String;
  attendance: Boolean;
}

export interface MeetResponse {
  ok: Boolean;
  data: Meet;
  msg: String;
}

export interface MeetAsistenResponse {
  data: {};
  msg: String;
  ok: Boolean;
}

export interface MeetPaginateResponse {
  totalPages: Number;
  data: Meet[];
}

export interface FilterMeet {
  show: Boolean;
  state?: string;
  date?: string;
}

export interface Reason {
  name: String;
  isActive: Boolean;
}
